sync/awsdatabricks.py

```diff
@@ -872,22 +860,56 @@ def _wait_for_cluster_termination(
     return Response(error=DatabricksAPIError(**cluster))
 
 
+def _cluster_log_destination(
+    cluster: dict,
+) -> Union[Tuple[str, str, str, str], Tuple[None, None, None, None]]:
+    cluster_log_conf = cluster.get("cluster_log_conf", {})
+    s3_log_url = cluster_log_conf.get("s3", {}).get("destination")
+    dbfs_log_url = cluster_log_conf.get("dbfs", {}).get("destination")
+
+    log_url = s3_log_url or dbfs_log_url
+
+    if log_url:
+        parsed_log_url = urlparse(log_url)
+        bucket = parsed_log_url.netloc
+        stripped_path = parsed_log_url.path.strip("/")
+
+        # If the event log destination is just a *bucket* without any sub-path, then we don't want to include
+        #  a leading `/` in our Prefix (which will make it so that we never actually find the event log), so
+        #  we make sure to re-strip our final Prefix
+        base_cluster_filepath_prefix = f"{stripped_path}/{cluster['cluster_id']}".strip("/")
+
+        return log_url, parsed_log_url.scheme, bucket, base_cluster_filepath_prefix
+
+    return None, None, None, None
```

Comment:
ParseResult might be a better return type: https://docs.python.org/3/library/urllib.parse.html#urllib.parse.ParseResult
```
>>> p = urlparse("s3://hello/world")
>>> p
ParseResult(scheme='s3', netloc='hello', path='/world', params='', query='', fragment='')
>>> p.geturl()
's3://hello/world'
```

Reply:
I was thinking that after looking at this again... I do want to return `base_cluster_filepath_prefix`, but maybe `Tuple[ParseResult, str]` would be better